plugins/git/diffview.nix

```diff
@@ -0,0 +1,328 @@
+            See ':h diffview-config-view.x.winbar_info'
+          '';
+        };
+
+        mergeTool = {
+          layout = mkStr "diff3_horizontal" ''
+            Config for conflicted files in diff views during a merge or rebase.
+            ${layoutsDescription}
+          '';
+
+          disableDiagnostics = mkBool true ''
+            Temporarily disable diagnostics for conflict buffers while in the view.
+          '';
+
+          winbarInfo = mkBool true ''
+            See ':h diffview-config-view.x.winbar_info'
+          '';
+        };
+
+        fileHistory = {
+          layout = mkStr "diff2_horizontal" ''
+            Config for changed files in file history views.
+            ${layoutsDescription}
+          '';
+
+          winbarInfo = mkBool false ''
+            See ':h diffview-config-view.x.winbar_info'
+          '';
+        };
+      };
+
+      filePanel = {
+        listingStyle = mkStr "tree" ''
+          One of 'list' or 'tree'
+        '';
+
+        treeOptions = let
+          commonDesc = "Only applies when listing_style is 'tree'";
+        in {
+          flattenDirs = mkBool true ''
+            Flatten dirs that only contain one single dir
+            ${commonDesc}
+          '';
+
+          folderStatuses = mkStr "only_folded" ''
+            One of 'never', 'only_folded' or 'always'.
+            ${commonDesc}
+          '';
+        };
+        winConfig = let
+          commonDesc = "See ':h diffview-config-win_config'";
+        in {
+          position = mkStr "left" commonDesc;
+
+          width = mkInt 35 commonDesc;
+
+          winOpts = mkAttributeSet commonDesc;
+        };
+      };
+      fileHistoryPanel = {
+        logOptions = let
+          commonDesc = "See ':h diffview-config-log_options'";
+        in {
+          git = {
+            singleFile = {
+              diffMerges = mkStr "combined" commonDesc;
+            };
+
+            multiFile = {
+              diffMerges = mkStr "first-parent" commonDesc;
+            };
+          };
+          hg = {
+            singleFile = mkAttributeSet commonDesc;
+
+            multiFile = mkAttributeSet commonDesc;
+          };
+        };
+        winConfig = let
+          commonDesc = "See ':h diffview-config-win_config'";
+        in {
+          position = mkStr "bottom" commonDesc;
+
+          height = mkInt 16 commonDesc;
+
+          winOpts = mkAttributeSet commonDesc;
+        };
+      };
+
+      commitLogPanel = {
+        winConfig = {
+          winOpts = mkAttributeSet ''
+            See ':h diffview-config-win_config'
+          '';
+        };
+      };
+
+      defaultArgs = let
+        commonDesc = "Default args prepended to the arg-list for the listed commands";
+      in {
+        diffviewOpen = mkNullable (types.listOf types.str) "[ ]" commonDesc;
+
+        diffviewFileHistory = mkNullable (types.listOf types.str) "[ ]" commonDesc;
+      };
+
+      hooks = mkAttributeSet ''
```

Comment:
Not many hooks are available, maybe list them all ?

Reply:
I listed them all, from CONTRIBUTING I understand that type for Lua code is mkStr right?

Reply:
If Lua code is the only possible value yes, else it can be an `either ..... rawType`